### PR TITLE
Introduce ThreadPoolExecutor without a Queue

### PR DESCRIPTION
**Pull request by fzakaria**
This adds the ability for a ThreadPoolExecutor to have a queue
depth of 0.

This is useful if you'd like to perform the rejection handler if no
available threads are availabile.

This is analogous to Java's use of SynchronousQueue in
ThreadPoolExecutor.

See https://stackoverflow.com/a/10186825/143733 for more details

### CODE REVIEW

**Comment by pitr-ch** [comment]
I had a brief look, NICE! I wanted to have this, good idea how to get around the problem with 0 already having a meaning.

**Comment by fzakaria** [comment]
Thanks. Look forward to your feedback.

**Comment by fzakaria** [comment]
@pitr-ch any update on this ?
I'd love to upgrade to this.

File: lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@
 require 'concurrent/concern/logging'
 require 'concurrent/executor/ruby_executor_service'
 require 'concurrent/utility/monotonic_time'
+require 'concurrent/mvar'
```
Comment (pitr-ch) [inline lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb]:
```suggestion
```
This is not used as far as I've seen.

File: lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb [hunk 2]
```diff
@@ -201,6 +210,10 @@ def ns_assign_worker(*args, &task)
     #
     # @!visibility private
     def ns_enqueue(*args, &task)
+      if @synchronous
+        return false
+      end
```
Comment (pitr-ch) [inline lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb]:
```suggestion
      return false if @synchronous
```

**Comment by pitr-ch** [comment]
I've checked and fixed documentation. I had to rebase on master.

**Comment by fzakaria** [comment]
Thanks @pitr-ch 
I added CHANGELOG.md description.

**Pull request merged**